app/controllers/admin/application_controller.rb

```diff
@@ -0,0 +1,21 @@
+# All Administrate controllers inherit from this
+# `Administrate::ApplicationController`, making it the ideal place to put
+# authentication logic or other before_actions.
+#
+# If you want to add pagination or other controller-level concerns,
+# you're free to overwrite the RESTful controller actions.
+module Admin
+  class ApplicationController < Administrate::ApplicationController
+    before_action :authenticate_admin
+
+    def authenticate_admin
+      # TODO Add authentication logic here.
```

Comment:
I am afraid that anyone can enter the admin area for now. Am I right?

Reply:
Yes, you're right. I'm not sure how to use authentication here if I'm using device-jwt in my app - any idea how could I do that?

Reply:
For a classical web application, you need to use a solution based on cookies in the browser instead of tokens.Let's discuss it later in the :telephone_receiver: